server/app/services/question/types/QuestionType.java

```diff
@@ -40,6 +41,12 @@ public enum QuestionType {
 
   private final String label;
   private final Class<? extends Question> supportedQuestion;
+  private static final Map<String, QuestionType> irregularMappings =
```

Comment:
Can you explain why this is necessary?

Reply:
I have the same question. This is a pretty core class and I'm trying to understand what doesn't work with the new user case.

Reply:
Good question. The motivation is to render a Thymeleaf fragment inside J2HTML.

Short answer: PHONE.getLabel() -> "Phone Number" -> of("Phone Number") -> not found

Long answer can be traced in the larger PR: https://github.com/civiform/civiform/pull/7835/files

1. Thymeleaf is loaded by HTMX via QuestionEditView.
2. `htmx-get` calls `controllers.admin.routes.NorthStarQuestionController.sampleQuestion(...)`
3. The method signature is `public Result sampleQuestion(Request request, String questionType)`, so we need a String to represent the question type
  * An enum cannot be used because htmx-get doesn't support enums
  * getLabel() is called to convert the enum to a String
4. NorthStarQuestionController recieves a String representing the question type.
5. The String needs to be converted back to an enum.
6. the `of(...)` method simply converts the String to uppercase. This misses conversions for some enums.

Reply:
Ok that makes sense. I think I'd prefer just rewriting that method to something like

```
public static QuestionType of(String name) throws InvalidQuestionTypeException {
    for (QuestionType type : QuestionType.values()) {
      if (type.getLabel().equals(name)) {
        return type;
      }
    }
    throw new InvalidQuestionTypeException(name);
  }
```